pkgs/applications/office/espanso/default.nix

```
@@ -108,47 +106,47 @@ rustPlatform.buildRustPackage rec {
 
   postPatch = lib.optionalString stdenv.isDarwin ''
     substituteInPlace scripts/create_bundle.sh \
-      --replace target/mac/ $out/Applications/ \
-      --replace /bin/echo ${coreutils}/bin/echo
+      --replace-fail target/mac/ $out/Applications/ \
+      --replace-fail /bin/echo ${coreutils}/bin/echo
     patchShebangs scripts/create_bundle.sh
     substituteInPlace espanso/src/res/macos/Info.plist \
-      --replace "<string>espanso</string>" "<string>${placeholder "out"}/Applications/Espanso.app/Contents/MacOS/espanso</string>"
-    substituteInPlace espanso/src/res/macos/com.federicoterzi.espanso.plist \
-      --replace "<string>/Applications/Espanso.app/Contents/MacOS/espanso</string>" "<string>${placeholder "out"}/Applications/Espanso.app/Contents/MacOS/espanso</string>" \
-      --replace "<string>/usr/bin" "<string>${placeholder "out"}/bin:/usr/bin"
+      --replace-fail "<string>espanso</string>" "<string>${placeholder "out"}/Applications/Espanso.app/Contents/MacOS/espanso</string>"
     substituteInPlace espanso/src/path/macos.rs  espanso/src/path/linux.rs \
-      --replace '"/usr/local/bin/espanso"' '"${placeholder "out"}/bin/espanso"'
+      --replace-fail '"/usr/local/bin/espanso"' '"${placeholder "out"}/bin/espanso"'
   '';
 
   # Some tests require networking
   doCheck = false;
 
-  postInstall = if stdenv.isDarwin then ''
-    EXEC_PATH=$out/bin/espanso BUILD_ARCH=current ${stdenv.shell} ./scripts/create_bundle.sh
-  '' else ''
-    wrapProgram $out/bin/espanso \
-      --prefix PATH : ${lib.makeBinPath (
-        lib.optionals stdenv.isLinux [
-          libnotify
-          setxkbmap
-        ] ++ lib.optionals waylandSupport [
-          wl-clipboard
-        ] ++ lib.optionals x11Support [
-          xclip
-        ]
-      )}
-  '';
+  postInstall =
+    if stdenv.isDarwin then ''
+      EXEC_PATH=$out/bin/espanso BUILD_ARCH=current ${stdenv.shell} ./scripts/create_bundle.sh
+    '' else ''
+      wrapProgram $out/bin/espanso \
+        --prefix PATH : ${lib.makeBinPath (
+          lib.optionals stdenv.isLinux [
+            libnotify
+            setxkbmap
+          ] ++ lib.optionals waylandSupport [
+            wl-clipboard
+          ] ++ lib.optionals x11Support [
+            xclip
+          ]
+        )}
+    '';
 
   passthru.tests.version = testers.testVersion {
     package = espanso;
+    # remove when updating to a release version
+    version = "2.2.1";
   };
 
   meta = with lib; {
     description = "Cross-platform Text Expander written in Rust";
     mainProgram = "espanso";
     homepage = "https://espanso.org";
     license = licenses.gpl3Plus;
-    maintainers = with maintainers; [ kimat pyrox0 ];
+    maintainers = with maintainers; [ kimat pyrox0 n8henrie ];
     platforms = platforms.unix;
```

```suggestion
    platforms = platforms.unix;
    broken = stdenv.hostPlatform.system == "x86_64-darwin";
```
since it seems to be broken on that system per CI

---

Thank you for pointing my at the CI failure. It references this symbol which was apparently added in the `apple_sdk_11_0` (or at least after the default sdk chosen for `x86_64-darwin`): [`kCFURLVolumeAvailableCapacityForImportantUsageKey`](https://developer.apple.com/documentation/corefoundation/kcfurlvolumeavailablecapacityforimportantusagekey). I'm now testing if a change in the SDK fixes the issue.

---

I currently see issues like this: 

```
cargo:warning=/nix/store/ck3awwmd2pca7cs50jbghcxfh32k1jc2-apple-framework-ColorSync-11.0.0/Library/Frameworks/ColorSync.framework/Headers/ColorSyncBase.h:96:69: note: expanded from macro 'CS_UNAVAILABLE_EMBEDDED'
  cargo:warning=#define CS_UNAVAILABLE_EMBEDDED API_UNAVAILABLE(ios, tvos, watchos, macCatalyst)
  cargo:warning=                                                                    

```

Will mark it as broken if I don’t find a solution.

---

I'm also seeing the error trying to compile sys-locale for x86_64-darwin.

I found [this issue](https://github.com/NixOS/nixpkgs/issues/214872) with a similar error fixed in [this PR](https://github.com/NixOS/nixpkgs/pull/215782) by @wegank . Unfortunately a similar change here (dropping `Foundation`, using `callPackage` from the 11.0 SDK, and explicitly inheriting frameworks from the 11.0 SDK) also fails.

Here's another open issue with a similar error: https://github.com/NixOS/nixpkgs/issues/210558 -- again @wegank comes to the rescue (with the suggestion to take dependencies from the 11.0 SDK).

---

I was informed a few hours ago (https://github.com/NixOS/nixpkgs/pull/312654#discussion_r1605794718) that the workaround actually stopped working since half a year ago, I have no idea what changed on the Rust side.

---

Marked x86_64-darwin as broken for now. Seems to be the best way forward to fix issues under linux with wayland and aarch64-darwin.